gym_wrapper/base_gym_env.py

```diff
@@ -0,0 +1,241 @@
+import os
+from typing import List, Optional
+
+import gym
+import numpy as np
+import pygame
+import vizdoom.vizdoom as vzd
+from gym import spaces
+from vizdoom import scenarios_path
+
+
+
+CONFIGS = [
+    ["basic.cfg", 3],  # 0
+    ["deadly_corridor.cfg", 7],  # 1
+    ["defend_the_center.cfg", 3],  # 2
+    ["defend_the_line.cfg", 3],  # 3
+    ["health_gathering.cfg", 3],  # 4
+    ["my_way_home.cfg", 5],  # 5
+    ["predict_position.cfg", 3],  # 6
+    ["take_cover.cfg", 2],  # 7
+    ["deathmatch.cfg", 20],  # 8
+    ["health_gathering_supreme.cfg", 3],  # 9
+]
+
+
+class VizdoomEnv(gym.Env):
+    def __init__(self, level, **kwargs):
+        """
+        Base class for Gym interface for ViZDoom. Thanks to https://github.com/shakenes/vizdoomgym
+        Child classes are defined in vizdoom_env_definitions.py,
+        that contain the level parameter and pass through any kwargs from gym.make()
+        :param level: index of level in the CONFIGS list above
+        :param kwargs: keyword arguments from gym.make(env_name_string, **kwargs) call. 'depth' will render the
+        depth buffer and 'labels' will render the object labels and return it in the observation.
+        Note that the observation will be a list with the screen buffer as the first element. If no kwargs are
+        provided (or depth=False and labels=False) the observation will be of type np.ndarray.
+        """
+
+        # parse keyword arguments
+        self.depth = kwargs.get("depth", False)
+        self.labels = kwargs.get("labels", False)
+        self.position = kwargs.get("position", False)
+        self.health = kwargs.get("health", False)
+
+        # init game
+        self.game = vzd.DoomGame()
+        self.game.set_screen_resolution(
+            vzd.ScreenResolution.RES_320X240
+        )  
+        self.game.load_config(os.path.join(scenarios_path, CONFIGS[level][0]))
+        self.game.set_window_visible(False)
+        self.game.set_depth_buffer_enabled(self.depth)
+        self.game.set_labels_buffer_enabled(self.labels)
+        self.game.clear_available_game_variables()
+        if self.position:
+            self.game.add_available_game_variable(vzd.GameVariable.POSITION_X)
+            self.game.add_available_game_variable(vzd.GameVariable.POSITION_Y)
+            self.game.add_available_game_variable(vzd.GameVariable.POSITION_Z)
+            self.game.add_available_game_variable(vzd.GameVariable.ANGLE)
+        if self.health:
+            self.game.add_available_game_variable(vzd.GameVariable.HEALTH)
+        self.game.init()
+        self.state = None
+        self.window_surface = None
+        self.isopen = True
+
+        self.action_space = spaces.Discrete(CONFIGS[level][1])
```

Comment:
You could infer this from the number of buttons enabled (`self.game.get_available_buttons_size()`)

However, you should probably check what available buttons there are and if they have `DELTA` buttons. Probably easier to skip supporting them for now and just throw `ValueError` telling that DELTA buttons are not supported.